pandas/_libs/tslibs/offsets.pyx

```diff
@@ -827,7 +823,55 @@ cpdef int get_day_of_month(datetime other, day_opt) except? -1:
         raise ValueError(day_opt)
 
 
-cpdef int roll_yearday(other, n, month, day_opt='start') except? -1:
+cpdef int _roll_convention(int other, int n, int compare):
```

Comment:
this is not very intutive, nor should you generally be calling this as a public function of a cython routine. name this something else or create separate functions so compare is more readable here. I don't even mind repeating some code, as trying to shove multiple functions into the same one is simply a bad idea here.

Reply:
I don't understand this comment.  This is in reference to _roll_convention?  I can rename it I guess, but I don't see how it could be separated into more specific functions.

Reply:
see below